### DISCUSSION TITLE
Can I provide theme thumbnails?

### DISCUSSION THREAD

**Discussion by jahnli** [rydmike/flex_color_scheme#204] [Q&A] [answered]
I want to give the user a choice of themes in my APP, but I have no idea how to display them, can I provide thumbnails of themes in themesplayground, I can't find them in the warehouse

![image](https://github.com/rydmike/flex_color_scheme/assets/20943608/fde20531-754b-43c0-b205-f23b0688a0d1)

**Comment by rydmike** [comment, marked as answer]
Hi @jahnli and thanks for your question.

If you want to you can certainly use the widget used above to make your own thumbnail theme selection too. The widget used to make the above thumbnail is actually a part of the package. Although it is not so difficult to build your own, but in very early version I was asked if I could add it as widget to the package, and I did and it is still around, just not very well advertised 😄   

In the docs you can find a bit info about it here: https://docs.flexcolorscheme.com/deep_dives#flexthememodeswitch

Where there are also links to its API docs. It was primary offered as a theme mode switch button: https://pub.dev/documentation/flex_color_scheme/latest/flex_color_scheme/FlexThemeModeSwitch-class.html but its building block just the box with the four colors is also available, https://pub.dev/documentation/flex_color_scheme/latest/flex_color_scheme/FlexThemeModeOptionButton-class.html

As shown in the docs page it can be styled quite a bit:

![ThemeSwitchCustomize](https://github.com/rydmike/flex_color_scheme/assets/39990307/1087c0e0-e0da-49e5-82ce-a713a8dd78f6)


For some examples of how to use it as theme selector, then talking about the [FlexThemeModeOptionButton](https://pub.dev/documentation/flex_color_scheme/latest/flex_color_scheme/FlexThemeModeOptionButton-class.html), 
you can check the open source **Themes Playground** code. The Playground is actually step/example 5, in FlexColorScheme usage tutorial: https://docs.flexcolorscheme.com/tutorial5

Here you can find the code for this full screen horizontal theme color selector 

![Screenshot 2023-10-19 at 15 11 39](https://github.com/rydmike/flex_color_scheme/assets/39990307/9de07129-0dab-4518-92e3-3ac08e6591a5)

and also for the vertical theme color selector

![Screenshot 2023-10-19 at 15 13 39](https://github.com/rydmike/flex_color_scheme/assets/39990307/9d3e69a4-468c-46ea-bc77-aaa4e480b304)


using the theme thumbs and see how it was used in the **Themes Playground** app:

https://github.com/rydmike/flex_color_scheme/blob/master/example/lib/example5_themes_playground/widgets/panels/theme_colors_settings/input_colors_selector.dart

In the **Themes Playground** colors panel, there is also a popup menu based theme selector:

> Since this code base was originally made as a tutorial for showing different possibilities it sometimes app wise contain different ways of modifying and doing things, just as a tutorial. As an app UX it can actually be a bit confusing, so I plan to remove these dual ways of doing things a future redesign of the Playground app.
  
![Screenshot 2023-10-19 at 14 48 52](https://github.com/rydmike/flex_color_scheme/assets/39990307/445ad2a1-52bf-4ffc-aeeb-a4df41e0e019)

In this case the code for the popup example is more "isolated" and it is a much easier example to use to see how it makes the theme color selection thumbs:

https://github.com/rydmike/flex_color_scheme/blob/master/example/lib/example5_themes_playground/widgets/panels/theme_colors_settings/input_colors_popup_menu.dart

This is just one way of making theme color thumb presentations, I've seen more elegant ways of providing a mini app as thumb with prominent colors shown on maybe 5 widgets. The thumbs used in this app is just one example that can also serve as a guide on how to make something even fancier.

**Reply by jahnli** [reply]
Thanks for your answer, I'll try later

**Reply by jahnli** [reply]
But there's almost no way to use a preset theme by naming it

FlexScheme。indigoM3, FlexScheme.material] . map ((v) = > FlexThemeModeOptionButton ())

@rydmike

**Reply by rydmike** [reply]
No, not directly from the `FlexScheme` scheme enum names to a button, but you can make such a mapping.

The `PopupMenuButton` builds the thumb buttons via a `FlexSchemeColor`. You you can get from the enum value to its `FlexSchemeColor` via the `FlexColor.schemes` map, and then the rest would be similar as the example. 

Extraction of the enum associated `FlexSchemeData` and from it the `FlexSchemeColor` colors directly via functions in the enum would be a handy feature though. I think that could be useful in the API. I will probably add that in the next release.

**Reply by rydmike** [reply]
Added an enhancement issue for the needed convenience API on enum `FlexScheme` here https://github.com/rydmike/flex_color_scheme/issues/210

**Reply by rydmike** [reply]
This fix/new feature for this is included the 8.0.0-dev release.

**FlexColorScheme** package dev release 8.0.0-dev.1 has been published. This is a BIG update with breaking changes and many new features. It has its own dev pre-release of the [**Themes Playground**](https://rydmike.com/flexcolorscheme/themesplayground-dev) as well, feel free to take a peak. No new API changes are planned for the stable 8.0.0 release from this dev.1 release.

* [Doc site has some pre-info](https://docs.flexcolorscheme.com/)
* [Dev release of PKG here](https://pub.dev/packages/flex_color_scheme/versions/8.0.0-dev.1)
* [The PKG changelog has the most detailed info](https://pub.dev/packages/flex_color_scheme/versions/8.0.0-dev.1/changelog)

**FOR INFO** there will be no new features or changes in it compared to this one. Only any issues found from the QA assurance will be addressed. The stable release will also include many doc updates, but docs site will continue to be edited and revised AFTER the 8.0.0 stable release.

**Reply by jahnli** [reply]
Thank you for your reply, when will v8 be released?

**Reply by rydmike** [reply]
You can use it already now if it is urgent. Even if it is 8.0.0-dev.1 release, there are no additional changes plans. The only diff between this 8.0.0-dev.1 and 8.0.0 release will be a lot of Quality Assurance work, which means reviewing and updating all existing tests and adding a lot of new test, plus also updating some doc API comments, although those are already up to date, but some double check.

One of the reasons for the pre-release/dev-release is also to check its pub scoring, which all look good.
 
![Screenshot 2024-10-14 at 10 16 09](https://github.com/user-attachments/assets/6ab2137b-e2b4-4ef6-96d8-0da7ab6a26cc)

Reviewing and doing all the test until they are up to my standard will take some time, no idea how long, sorry, it is all evening and weekend work :(

The doc site will be updated after the 8.0.0 stable release, gradually.

A dev release of the 8.0.0 Themes Playground is already out too: 
https://rydmike.com/flexcolorscheme/themesplayground-dev/

It has a lot of new features.

Once the stable 8.0.0 is ready I plan to to make some short form video tutorials too.

![Screenshot 2024-10-14 at 3 07 08](https://github.com/user-attachments/assets/642cb591-ea44-48d5-a98c-931c813d3691)

**Reply by jahnli** [reply]
Thank you for your work and look forward to it.
